### PR TITLE
Go Project Initialisation & Analyses tools

### PR DESCRIPTION
**Pull request by antlko**
1. Added Entry go "Hello World" files (for testing building and testing pipelines)
2. Added CodeQL analyser
3. Added SonarCloud code analyzer
4. Added Dependabot config to contain fresh dependencies in future

For configuring SonarCloud:

1. @kirillt has to create organisation in https://sonarcloud.io/account/organizations and check this repo
3. @kirillt has to add to secrets SONAR_TOKEN parameter
4. Have to be updated sonarcloud.properties file with the required project data

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module github.com/ARK-Builders/ark-p2p
```
Comment (antlko) [inline go.mod:1]: Module named as group name

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,17 @@
+sonar.projectKey=antlko_ark-p2p
```
Comment (antlko) [inline sonar-project.properties:1]: Will be changed after the creating organisation in [sonarcl](https://sonarcloud.io/)

File: go.mod (hunk 1 above)
Comment (kirillt) [reply go.mod:1]: Should be fine

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+sonar.organization=antlko
+
```
[the remark this answers is not in the document]
Comment (kirillt) [reply sonar-project.properties:3]: Created `ark-builders` organization: https://sonarcloud.io/organizations/ark-builders/projects
`SONAR_TOKEN` is initialized, too.

**Pull request merged**